### PR TITLE
enable env variable on pimRequirement and console for DB

### PR DESCRIPTION
**Pull request by sebglon**
**Description (for Contributor and Core Developer)**
The goal is add possibility to config Mysql connection with environment variable.
This is a merge with onboarder cloud stuff

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Comment by AymericPlanche** [comment]
i think you have to provide a .env-dist file at the root of the project that contains the default env variables and update the readme / installation guide.

File: .ci/Dockerfile [hunk 1]
```diff
@@ -71,7 +71,8 @@ COPY . .
 COPY .ci/php.ini /usr/local/etc/php/
 COPY .ci/vhost.conf /etc/apache2/sites-available/000-default.conf
 
-RUN composer $COMPOSER_COMMAND --no-ansi --optimize-autoloader --no-interaction --no-progress --prefer-dist --no-suggest \
+RUN cp app/config/parameters.yml.dist app/config/parameters.yml \
```
Comment (ahocquard) [inline .ci/Dockerfile]: parameters.yml.dist is useless now, we should delete it

Comment (sebglon) [reply .ci/Dockerfile]: no it is still used by clearCache
https://ci.akeneo.com/blue/organizations/jenkins/akeneo%2Fpim-community-dev/detail/PR-8597/1/pipeline#step-40-log-1368

If you can help me to remove it, I would appreciate your help

File: app/config/parameters.yml.dist [hunk 2]
```diff
@@ -1,13 +1,24 @@
 parameters:
+    env(PIM_ELASTICSEARCH_HOST): "localhost"
+    env(PIM_ELASTICSEARCH_PORT): "9200"
+    env(PIM_ELASTICSEARCH_USER): "elastic"
```
Comment (damien-carcel) [inline app/config/parameters.yml.dist]: Beware, most of local installs of elasticsearch don't have any login/password.

**Comment by sebglon** [comment]
Can we retry to work on this for the 3.X. 
Using Env variable is mandatory for the serenity offer

**Comment by skeleton** [comment]
do you really thing it's the time to do it? As you may know, we freeze the master today.

**Comment by sebglon** [comment]
yes but same than 2.3 cloud product is not included on the release process of the pim.

**Comment by sebglon** [comment]
For me the CI is good except the same test fail than master and timeout

**Comment by jjanvier** [comment]
Do we need to do the same for EE and for standard editions?
Is it normal the .env.dist is not updated?

File: bin/console [hunk 3]
```diff
@@ -25,6 +26,11 @@ if ($debug) {
     Debug::enable();
 }
 
+$envFile = __DIR__ . '/../.env'.(($env != 'prod') ? '-'.$env: '');
+if (file_exists($envFile)) {
+    (new Dotenv())->load($envFile);
```
Comment (arnolanglade) [inline bin/console]: You should load env file in every front controller (app_dev.php / app_behat.php, etc.)

**Comment by BitOne** [comment]
Obsolete since using Symfony 4.

**Pull request closed without merging**